plugins/base.py

```diff
@@ -2,15 +2,28 @@
 from typing import Callable, List, Tuple, Any
 
 from blocks.pipeline import Pipeline
+from configs.constants import LogConst
 
 from utils.random import random_string
 from runner.store import Store
 import pandas as pd
 
 
+def just_custom_functions(obj) -> List[Tuple]:
+    return [func for func in vars(obj).items() if not func[0].startswith("__")]
+
+
 class Plugin(ABC):
     def __init__(self):
-        self.id = f"{self.__class__.__name__} - {random_string(5)}"
+        pass
+
+    def __init_subclass__(cls):
+        cls.id = f"{cls.__name__} - {random_string(5)}"
```

Comment:
why do we need this in the class? can't we handle this outside, in the runner?

Reply:
This is imo a neater solution. The id is set in other places in the class itself aswell. Moving it from ``__init__`` where it was originally to ``__init__subclass__`` just ensures that there is an init even if a plugin overwrites the ``__init__`` function and then doesn't call ``super().__init()__``

Reply:
don't agree! as much business logic in runner, as little in the base class (preferably zero)